<launch>
    <!-- Toggle if using simulation data -->
    <param name="/use_sim_time" value="true"/>

    <node pkg="nodelet"
          type="nodelet"
          name="rgb_to_hsv"
          args="load sb_pointcloud_processing/rgb_to_hsv nm" output="screen">
        <remap from="~/input" to="/zed/point_cloud/cloud_registered" />
    </node>

    <node pkg="nodelet"
          type="nodelet"
          name="h"
          args="load pcl/PassThrough nm" output="screen">
        <remap from="~input" to="rgb_to_hsv/output" />
        <rosparam>
            filter_field_name: h
            filter_limit_min: 0
            filter_limit_max: 360
            filter_limit_negative: False
        </rosparam>
    </node>

    <node pkg="nodelet"
          type="nodelet"
          name="s"
          args="load pcl/PassThrough nm" output="screen">
        <remap from="~input" to="/h/output" />
        <rosparam>
            filter_field_name: s
            filter_limit_min: 0
            filter_limit_max: 1
            filter_limit_negative: False
        </rosparam>
    </node>

    <node pkg="nodelet"
          type="nodelet"
          name="v"
          args="load pcl/PassThrough nm" output="screen">
        <remap from="~input" to="/s/output" />
        <rosparam>
            filter_field_name: v
            filter_limit_min: 0
            filter_limit_max: 1
            filter_limit_negative: False
        </rosparam>
    </node>

    <node pkg="nodelet"
          type="nodelet"
          name="z"
          args="load pcl/PassThrough nm" output="screen">
        <remap from="~input" to="/v/output" />
        <rosparam>
            filter_field_name: z
            filter_limit_min: -10
            filter_limit_max: 10
            filter_limit_negative: False
            input_frame: zed_left_camera
            output_frame: zed_left_camera
        </rosparam>
    </node>

</launch>
